File: client_java/src/main/java/de/tron/client_java/network/message/Message.java
package de.tron.client_java.network.message;

import java.util.ArrayList;
import java.util.List;

public class Message {
	
	int lobbyId = -1;
	int length = -1;
	
	private MessageType type;
	private Action action;
	
	private final List<Player> players = new ArrayList<>();

	public int getLobbyId() {
		return lobbyId;
	}

	public void setLobbyId(int lobbyId) {
		this.lobbyId = lobbyId;
	}
	
	public int getLength() {
		return length;
	}

	public void setLength(int length) {
		this.length = length;
	}

	public MessageType getType() {
		return type;
	}

	public void setType(MessageType type) {
		this.type = type;
	}

	public Action getAction() {
		return action;
	}

	public void setAction(Action action) {
		this.action = action;
	}

	public List<Player> getPlayers() {
		return players;
	}

}

File: client_cpp/tron_client/src/message/message.cpp
#include "message.h"

void to_json(json& j, const Message& m)
{
    j = json({
        {"type", m.type},
        {"action", m.action},
        {"lobbyId", m.lobbyId},
        {"length", m.length},
        {"players", m.players}
    });

}

void from_json(const json& j, Message& m)
{
    j.at("type").get_to(m.type);
    j.at("action").get_to(m.action);
    j.at("lobbyId").get_to(m.lobbyId);
    j.at("length").get_to(m.length);
    j.at("players").get_to(m.players);
}
